cql2pgjson/src/main/java/org/folio/cql2pgjson/util/DbSchemaUtils.java

```diff
@@ -54,4 +65,99 @@ public static Index getIndex(String cqlIndex, List<Index> indexes) {
+    for (Table table : dbSchema.getTables()) {
+      if (tableName.equals(table.getTableName())) {
+        return table;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Find a list of {@link ForeignKeys} from source table to target table alias
+   *
+   * @param dbSchema
+   * @param srcTabName
+   * @param targetTabAlias
+   *
+   * @return a list of {@link ForeignKeys}
+   */
+  public static List<DbFkInfo> findForeignKeysFromSourceTableToTargetAlias(Schema dbSchema, String srcTabName,
+      String targetTabAlias) {
+    return findForeignKeys(dbSchema, srcTabName, targetTabAlias, true);
+  }
+
+  private static List<DbFkInfo> findForeignKeys(Schema dbSchema, String srcTabName, String targetTabName,
+      boolean useTargetAlias) {
+    List<DbFkInfo> list = new ArrayList<>();
+    Table srcTab = getTable(dbSchema, srcTabName);
+    if (srcTab.getForeignKeys() == null) {
+      return list;
+    }
+    // direct FK
+    for (ForeignKeys fk : srcTab.getForeignKeys()) {
+      String targetName = useTargetAlias ? fk.getTargetTableAlias() : fk.getTargetTable();
+      if (targetTabName.equals(targetName)) {
+        list.add(new DbFkInfo(srcTab.getTableName(), fk.getFieldName(), fk.getTargetTable()));
+        return list;
+      }
+    }
+    // find the shortest path
+    for (ForeignKeys fk : srcTab.getForeignKeys()) {
+      updateFkList(list, dbSchema, srcTab, fk, targetTabName, useTargetAlias);
+    }
+    return list;
+  }
+
+  private static void updateFkList(List<DbFkInfo> list, Schema dbSchema, Table srcTab, ForeignKeys fk,
+      String targetTabName, boolean useTargetAlias) {
+    List<DbFkInfo> childList = findForeignKeys(dbSchema, fk.getTargetTable(), targetTabName, useTargetAlias);
+    if (!childList.isEmpty()) {
+      if (!list.isEmpty() && (list.size() > (childList.size() + 1))) {
+        list.clear();
+      }
+      list.add(new DbFkInfo(srcTab.getTableName(), fk.getFieldName(), fk.getTargetTable()));
+      list.addAll(childList);
+    }
+  }
+
+  /**
+   * Find a list of {@link ForeignKeys} from source table alias and target table.
+   *
+   * @param dbSchema
+   * @param srcTabAlias
+   * @param targetTabName
+   * @return
+   */
+  public static List<DbFkInfo> findForeignKeysFromSourceAliasToTargetTable(Schema dbSchema, String srcTabAlias,
+      String targetTabName) {
+    List<DbFkInfo> list = new ArrayList<>();
+    for (Table table : dbSchema.getTables()) {
+      if (table.getForeignKeys() == null) {
+        continue;
+      }
+      for (ForeignKeys fk : table.getForeignKeys()) {
+        String tabAlias = fk.getTableAlias() == null ? table.getTableName() : fk.getTableAlias();
+        if (srcTabAlias.equals(tabAlias)) {
+          // direct FK
+          if (targetTabName.equals(fk.getTargetTable())) {
+            list.add(new DbFkInfo(table.getTableName(), fk.getFieldName(), fk.getTargetTable()));
+            return list;
+          } else {
+            // find the shortest path
```

Comment:
Please explain which one is chosen when two paths have the same length:
tableA -> tableX -> tableB
tableA -> tableY -> tableB

Reply:
Same here.